docs/rules/no-get-by-assert.md

```diff
@@ -0,0 +1,51 @@
+# Disallow `getBy*` queries as assertion method itself (no-get-by-assert)
+
+Testing Library `getBy*` queries throw an error if the element is not
+found. Some users like this behavior to use the query itself as an
+assert for the element existence in their tests, but other users don't
+and prefer to explicitly assert the element existence, so this rule is
+for users from the latter.
+
+## Rule Details
+
+This rule aims to encourage users to explicitly assert existence of
+elements in their tests rather than just use `getBy*` queries and expect
+it doesn't throw an error so it's easier to understand what's the
+expected behavior within the test.
+
+Examples of **incorrect** code for this rule:
+
+```js
+// just calling `getBy*` query expecting not to throw an error as an
+// assert-like method, without actually using the returned element
+getByText('foo');
+```
+
+Examples of **correct** code for this rule:
+
+```js
+// wrapping the get query within a `expect` and use some matcher for
+// making the assertion more explicit
+expect(getByText('foo')).toBeDefined();
```

Comment:
Shouldn't it be `expect(queryByText)` instead of `expect(getByText)`? 🤔 
Such an example would be conflicting with the `prefer-expect-query-by` rule.

Reply:
That's actually from a different rule so I would avoid having this one encouraging two things: asserting explicitly and asserting with `queryBy*`, so just leaving this rule for the former and `prefer-expect-query-by` for the latter.

Reply:
Yes, but as `prefer-expect-query-by` is in the recommended rules, I found it a little bit weird to say `expect(getBy)` is correct (although it is for this rule technically speaking) 🙂

Reply:
Well, in the example of this specific rule that's fine. It would be weird to see `expect(queryByText('foo')).toBeDefined();` when the rule only talk about `getBy*`, so I wanted to clarify that specific use is valid for this rule.

Reply:
Sure! What about putting a side note stating that though `expect(getBy)` is valid for this rule, it is recommended to use `queryBy` for asserting elements' presence? Then, the user would know it's valid to use `expect(getBy)` while still being aware it's not recommended. What do you think? 🙂

Reply:
That sounds absolutely reasonable. I'll address the feedback here later.

Reply:
My 2 cents: I understand both points. However, since this is a documentation to something that is to be considered "valid", I would expect that we write things that are "valid", which is not really the case with `expect(getBy)`, as we also have a recommended rule for it.

At least we should mention to wrap it into an `expect` and then to change it to `queryBy`, and link it to the other rule for more reference.